nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -113,102 +116,156 @@ def __init__(
         self.post_process = post_process
         self.fp16_cross_entropy = fp16_cross_entropy
         self.precision = precision
+        self.add_encoder = add_encoder
```

Comment:
Are we preparing this class to support encoder or decoder only models (so we can disable either)?

Reply:
No, this is PP specific since certain GPUs may have encoder only or decoder only.